// 首行指定proto语法版本，默认选用proto2
syntax = "proto3";


/*
定义传入的request字段
*/
message request {
    string user_id=1;
    int32 age=2;
    int32 gender=3;
    int32 platform=4;
    int32 video_nums=5;
}

/*
定义返回的video reponse字段
*/
message Meta {
    string cover=1;
    string title=2;
    string up=3;
    string tag=4;
}
message Video {
    int64 video_id=1;
    Meta Meta=2;
}
message VideoResponse {
    string impression=1;
    int64 time_stamp=2;
    repeated Video rec_algo=3;
}

/*
对request和response建立gRPC服务链接
*/
service RecSystem {
    rpc rec_sys(request) returns(VideoResponse) {
    }
}

//end